contracts/rewards/src/contract/execute.rs

```diff
@@ -77,113 +83,110 @@ where
     ) -> Result<(), ContractError> {
         let cur_epoch = self.current_epoch(block_height)?;
 
+        let event =
+            self.load_or_store_event(event_id, contract_addr.clone(), cur_epoch.epoch_num)?;
+
+        self.store
+            .load_epoch_tally(contract_addr.clone(), event.epoch_num)?
+            .unwrap_or(EpochTally::new(
+                contract_addr,
+                cur_epoch,
+                self.store.load_params().params,
+            ))
+            .record_participation(worker)
+            .then(|mut tally| {
+                if matches!(event, LoadState::New(_)) {
```

Comment:
The enum name is confusing. I'd prefer renaming `Loaded` to `Existing` and `LoadState` to just `State`.